module/module_test.go

```diff
@@ -493,192 +463,124 @@ func TestModularResourceWithoutReconfigure(t *testing.T) {
-			var conf MockSummationConfig
-			return config.TransformAttributeMapToStruct(&conf, attributes)
-		},
-		&MockSummationConfig{})
+		_, err = m.ReconfigureResource(ctx, &pb.ReconfigureResourceRequest{
+			Config: mockConf, Dependencies: deps,
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	sumAttrs, err := protoutils.StructToStructPb(MockSummationConfig{
-		Motors: []string{motor.Named("motor1").String()},
+		test.That(t, createDeps1, test.ShouldResemble, deps)
+		test.That(t, createConf1.Attributes, test.ShouldResemble, mockConf.Attributes)
+		test.That(t, createConf1.ConvertedAttributes, test.ShouldImplement, &MockConfig{})
 	})
-	test.That(t, err, test.ShouldBeNil)
-	sumConf := &v1.ComponentConfig{
-		Name:       "mymocksummation1",
-		Api:        summationapi.Subtype.String(),
-		Model:      model.String(),
-		Attributes: sumAttrs,
-	}
 
-	newSumAttrs, err := protoutils.StructToStructPb(MockSummationConfig{
-		Motors: []string{motor.Named("motor2").String()},
-	})
-	test.That(t, err, test.ShouldBeNil)
-	newSumConf := &v1.ComponentConfig{
-		Name:       "mymocksummation1",
-		Api:        summationapi.Subtype.String(),
-		Model:      model.String(),
-		Attributes: newSumAttrs,
-	}
+	t.Run("reconfigurable creation", func(t *testing.T) {
+		mockAttrs, err := protoutils.StructToStructPb(MockConfig{
+			Motors: []string{motor.Named("motor1").String()},
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	// Add the mockSummation model to the module registry
-	test.That(t, m.AddModelFromRegistry(ctx, summationapi.Subtype, model),
-		test.ShouldBeNil)
+		mockReconfigConf.Attributes = mockAttrs
+		mockReconfigConf.Model = modelWithReconfigure.String()
 
-	test.That(t, m.Start(ctx), test.ShouldBeNil)
+		validateResp, err := m.ValidateConfig(ctx, &pb.ValidateConfigRequest{
+			Config: mockReconfigConf,
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	conn, err := grpc.Dial(
-		"unix://"+addr,
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
-		grpc.WithStreamInterceptor(grpc_retry.StreamClientInterceptor()),
-		grpc.WithUnaryInterceptor(grpc_retry.UnaryClientInterceptor()),
-	)
-	test.That(t, err, test.ShouldBeNil)
+		deps := validateResp.Dependencies
+		test.That(t, deps, test.ShouldResemble, []string{"rdk:component:motor/motor1"})
 
-	client := pb.NewModuleServiceClient(conn)
-	m.SetReady(true)
-	readyResp, err := client.Ready(ctx, &pb.ReadyRequest{ParentAddress: parentAddr})
-	test.That(t, err, test.ShouldBeNil)
-	test.That(t, readyResp.Ready, test.ShouldBeTrue)
+		_, err = m.AddResource(ctx, &pb.AddResourceRequest{
+			Config: mockReconfigConf, Dependencies: deps,
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	validateResp, err := m.ValidateConfig(ctx, &pb.ValidateConfigRequest{
-		Config: sumConf,
+		test.That(t, reconfigDeps1, test.ShouldResemble, deps)
+		test.That(t, reconfigConf1.Attributes, test.ShouldResemble, mockReconfigConf.Attributes)
+		test.That(t, reconfigConf1.ConvertedAttributes, test.ShouldImplement, &MockConfig{})
```

Comment:
Shouldn't this test the actual values within ConvertedAttributes? Not just that it implements MockConfig{}? Zeroed out MockConfig{} would pass this test but not be valid behavior.